### ISSUE TITLE
Request for shoulder swap function

### ISSUE THREAD

**Issue by greenleaf1mile** [mwilsnd/SkyrimSE-SmoothCam#34]
Can you add a feature that makes the shoulder swap function automatic when certain conditions are met? The certain condition I have in mind are when the X and - Y distance clamps are hit. This would give you a more smooth camera without the actor taking up the center of the screen when the camera is acting lazy (for high depth of field when using a ENB)

I have been trying to make a over the shoulder preset. With the min smoothing rate set to 1.00 so the camera is fixed over the shoulder and with 1 scroll out, and the max smoothing rate set to .01, the camera is lazy. With the correct min/max distances you have 3 different camera setting with only 3 zoom settings (first person, over the shoulder, and then you have a lazy camera with the third scroll out) but the only way I can achieve this is if I de-activate the local space interpolation. This causes an issue when the camera is acting "lazy" at the max distance. The camera rotates in place and the charter will clip out of the screen.This cannot be balanced out with the smoothing rate. 

With the FOV feature you have added, it makes over the shoulder gameplay cinematic af . I will be releasing a preset soon that takes advantage of the FOV.

**Comment by mwilsnd** [comment]
I've started playing around with this and have an early version working when the camera crosses the X axis limits, it needs more work though. I'll let you know more when I'm happy with the functionality. I also have some ideas for a method to automate the distance clamping ranges such that the player can never run off the sides of the screen which should work well for your Y axis case, I might try and work on that later.

**Comment by greenleaf1mile** [comment]
Yes, thanks for the update. I can test anything out if you need a second opinion. In the meantime, I will be posting any bug but as far as I can tell, this camera is pretty flawless.

Thanks for the amazing mod.

**Comment by lovelydumpling** [comment]
I know this is ancient but I second this request. As a controller player without any real room for another button to dedicate to shoulder swapping it makes it difficult for me to use like 90% of smoothcam presets because almost everyone aims for a shoulder style.
